components/script/dom/response.rs

```diff
@@ -142,80 +144,27 @@ fn is_null_body_status(status: u16) -> bool {
 }
 
 impl ResponseMethods<crate::DomTypeHolder> for Response {
-    // https://fetch.spec.whatwg.org/#initialize-a-response
+    // https://fetch.spec.whatwg.org/#dom-response
     fn Constructor(
         global: &GlobalScope,
         proto: Option<HandleObject>,
         can_gc: CanGc,
-        body: Option<BodyInit>,
+        body_init: Option<BodyInit>,
         init: &ResponseBinding::ResponseInit,
     ) -> Fallible<DomRoot<Response>> {
-        // Step 1
-        if init.status < 200 || init.status > 599 {
-            return Err(Error::Range(format!(
-                "init's status member should be in the range 200 to 599, inclusive, but is {}",
-                init.status
-            )));
-        }
-
-        // Step 2
-        if !is_valid_status_text(&init.statusText) {
-            return Err(Error::Type(
-                "init's statusText member does not match the reason-phrase token production"
-                    .to_string(),
-            ));
-        }
-
+        // 1. Set this’s response to a new response.
```

Comment:
It would be interesting to add a note, and perhaps open an issue, to note the fact that here we are conflating a dom response with the fetch response concept, which is dealt with at https://github.com/servo/servo/blob/5e2d42e94459af76c8752fef934a7dde8ac5b41a/components/shared/net/response.rs#L94

I think that may be also why the "else" steps are necessary at the end of `initialize_response`, and we're going to have to deal with that some day.

Reply:
Ah, that actually clears things up for me a little bit as well.